packages/babel-parser/src/plugins/jsx/index.js

```
@@ -507,13 +507,35 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       return this.jsxParseElementAt(startPos, startLoc);
     }
 
+    // Validates and parses the current state as JSXText
+
+    parseJSXText(): N.Expression {
+      const startPos = this.state.start;
+      for (let idx = 0; idx < this.state.value.length; idx += 1) {
+        const character = this.state.value[idx];
+        if (character === "}") {
```

Do we also need to need to throw an error for `{` and `<`?

---

Ah, ignore me - this already throws.